packages/react-art/src/__tests__/ReactART-test.js

```diff
@@ -329,3 +334,103 @@ describe('ReactART', () => {
+        'Warning: Failed prop type: The prop `radius` is marked as required in `Circle`, ' +
+          'but its value is `undefined`.' +
+          '\n    in Circle (at **)',
+      );
+    }
+  });
+
+  it('should generate a Shape module with props for drawing the Rectangle', () => {
+    const rectangle = renderer.create(
+      <Rectangle width={50} height={50} stroke="green" fill="blue" />,
+    );
+    expect(rectangle.toJSON()).toMatchSnapshot();
+  });
+
+  it('should warn if width/height is missing on a Rectangle component', () => {
+    spyOnDev(console, 'error');
+    renderer.create(<Rectangle stroke="green" fill="blue" />);
+    if (__DEV__) {
+      expect(console.error.calls.count()).toBe(2);
+      expect(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toEqual(
+        'Warning: Failed prop type: The prop `width` is marked as required in `Rectangle`, ' +
+          'but its value is `undefined`.' +
+          '\n    in Rectangle (at **)',
+      );
+      expect(normalizeCodeLocInfo(console.error.calls.argsFor(1)[0])).toEqual(
+        'Warning: Failed prop type: The prop `height` is marked as required in `Rectangle`, ' +
+          'but its value is `undefined`.' +
+          '\n    in Rectangle (at **)',
+      );
+    }
+  });
+
+  it('should generate a Shape module with props for drawing the Wedge', () => {
+    const wedge = renderer.create(
+      <Wedge outerRadius={50} startAngle={0} endAngle={360} fill="blue" />,
+    );
+    expect(wedge.toJSON()).toMatchSnapshot();
+  });
+
+  it('should warn if outerRadius/startAngle/endAngle is missing on a Wedge component', () => {
+    spyOnDev(console, 'error');
+    renderer.create(<Wedge fill="blue" />);
+    if (__DEV__) {
+      expect(console.error.calls.count()).toBe(3);
+      expect(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toEqual(
+        'Warning: Failed prop type: The prop `outerRadius` is marked as required in `Wedge`, ' +
+          'but its value is `undefined`.' +
+          '\n    in Wedge (at **)',
+      );
+      expect(normalizeCodeLocInfo(console.error.calls.argsFor(1)[0])).toEqual(
+        'Warning: Failed prop type: The prop `startAngle` is marked as required in `Wedge`, ' +
+          'but its value is `undefined`.' +
+          '\n    in Wedge (at **)',
+      );
+      expect(normalizeCodeLocInfo(console.error.calls.argsFor(2)[0])).toEqual(
+        'Warning: Failed prop type: The prop `endAngle` is marked as required in `Wedge`, ' +
+          'but its value is `undefined`.' +
+          '\n    in Wedge (at **)',
+      );
+    }
+  });
+
+  it('should turn degrees to radians', () => {
+    const wedgeInstance = renderer
+      .create(
+        <Wedge outerRadius={50} startAngle={0} endAngle={360} fill="blue" />,
+      )
+      .getInstance();
+    expect(wedgeInstance._degreesToRadians(0)).toBe(0);
+    expect(wedgeInstance._degreesToRadians(180)).toBe(3.141592653589793);
```

Comment:
I don’t think we should be calling internal functions. Just using props is enough.

Reply:
ok, these tests are removed.
Can't find a good way to test the math generating `path`, all other props are covered in snapshot tests.
Please let me know if anything's missing.